### PR TITLE
Ev 13

### PR DESCRIPTION
**Pull request by Snehithlal**

### CODE REVIEW

File: app/controllers/api/v1/events_controller.rb [hunk 1]
```diff
@@ -13,11 +15,36 @@ def create
...
+        render json: EventSerializer.render_as_json(@event, root: :event, view: :with_all_associations),
+          status: :ok
+      end
+
+      def update
+        if @event.update(event_params)
+          render json: EventSerializer.render_as_json(@event, root: :event, view: :with_all_associations),
+          status: :ok
+        else
+          render json: { errors: @event.errors.full_messages },
```
Comment (anbk77) [inline app/controllers/api/v1/events_controller.rb:33]: better to use errors.messages? @Snehithlal

File: app/models/user_event.rb [hunk 2]
```diff
@@ -4,5 +4,5 @@ class UserEvent < ApplicationRecord
   belongs_to :event
   belongs_to :user
 
-  validates :user_id, presence: true
+  validates :user_id, presence: true, uniqueness: { scope: :event_id }
```
Comment (anbk77) [inline app/models/user_event.rb:7]: see if uniquness works when bulk creation or updation with nested attributes

Comment (Snehithlal) [reply app/models/user_event.rb:9]: it should work right?

File: app/models/event.rb [hunk 3]
```diff
@@ -15,11 +14,16 @@ class Event < ApplicationRecord
...
+  before_create :add_host_to_the_user_events
 
   private
 
   def validate_end_time_after_start_time
     errors.add(:end_time, 'cant be before start time') if end_time <= start_time
   end
+
+  def add_host_to_the_user_events
+    user_events.build(user_id: host_id, event_role: 'admin')
```
Comment (anbk77) [inline app/models/event.rb:27]: @Snehithlal should be 1

**Comment by Snehithlal** [comment]
[[EV-13] Add Event Edit And  Update Page](https://trello.com/c/edJoO13U/15-ev-13-add-event-edit-and-update-page)

**Pull request merged**